store/root.js

```diff
@@ -105,26 +103,33 @@ class Root extends Store {
   }
 
   @computed
-  get dataProviders() {
-    return this.user?.dataProviders
+  get dataProvider() {
+    return this.user.dataProvider
+  }
+
+  set dataProvider(dataProvider) {
+    // Check access rights
+    if (!this.user.canManage(dataProvider.id)) {
```

Comment:
Here should be another condition kind of `!this.options.allowAnonymousAccess && ...` but it can be added in the future.

Also, it would be good it the `use.canManage` check the user's permissions, it the user is an admin, it should always return `true`

Reply:
Yeah truth but allowAnonymousAccess doesn't work yet. So it can be added later as you said. The same for the second paragraph.